homeassistant/components/zwave/__init__.py

```diff
@@ -712,8 +720,9 @@ def _value_handler(self, method=None, class_id=None, index=None,
         May only be used inside callback.
 
         """
-        if class_id is not None:
+        if class_id is not None and not isinstance(class_id, list):
```

Comment:
The code surrounding `class_id` is very complex:
 - if `class_id` is given but it is not a list, use OZW method to filter it
 - if `class_id` is given but it is a list, use our methods to filter it.

I think that this should be done in a nicer way:

```python
values = []
if not isinstance(class_id, list):
    class_id = [class_id]
for cid in class_id:
    values.extend(self._value.node.get_values(class_id=cid).values())
```

And now you don't need to add any extra filtering below.